packages/react/src/components/SuiteHeader/SuiteHeaderAppSwitcher/SuiteHeaderAppSwitcher.test.jsx

```diff
@@ -109,13 +108,57 @@ describe('SuiteHeaderAppSwitcher', () => {
   it('shows loading state', async () => {
     delete window.location;
     window.location = { href: '' };
-    const wrapper = mount(
+    render(
       <SuiteHeaderAppSwitcher
         allApplicationsLink="https://www.ibm.com"
         noAccessLink="https://www.ibm.com"
       />
     );
     // Expect skeletons
-    expect(wrapper.find('[data-testid="suite-header-app-switcher--loading"]')).toHaveLength(1);
+    expect(screen.getByTestId('suite-header-app-switcher--loading')).toBeVisible();
+  });
+
+  it('calls the default onRouteChange', async () => {
+    jest.spyOn(SuiteHeaderAppSwitcher.defaultProps, 'onRouteChange');
```

Comment:
nit: just a minor reflection. I think that it is not actually the call to  `SuiteHeaderAppSwitcher.defaultProps.onRouteChange` that we are interested in since it always returns true no matter what we pass in. Instead, we probably want to make sure that we get a certain functionality even if we don't supply a `onRouteChange` prop.

Reply:
fair point. added a expect on the href for this.